src/pragha-utils.c

```diff
@@ -233,30 +233,30 @@ gchar* convert_length_str(gint length)
 	str = g_new0(char, 128);
 	memset(tmp, '\0', 24);
 
-	if (length > 86400) {
+	if (length >= 86400) {
 		days = length/86400;
 		length = length%86400;
 		g_sprintf(tmp, "%d %s, ", days, ngettext("day", "days", days));
 		g_strlcat(str, tmp, 24);
 	}
 
-	if (length > 3600) {
+	if (length >= 3600) {
 		hours = length/3600;
 		length = length%3600;
 		memset(tmp, '\0', 24);
 		g_sprintf(tmp, "%d:", hours);
 		g_strlcat(str, tmp, 24);
 	}
 
-	if (length > 60) {
+	if (length >= 60) {
 		minutes = length/60;
 		length = length%60;
 		memset(tmp, '\0', 24);
 		g_sprintf(tmp, "%02d:", minutes);
 		g_strlcat(str, tmp, 24);
 	}
 	else
-		g_strlcat(str, "00:", 4);
+		g_strlcat(str, "00:", 24);
```

Comment:
I assumed that you used `4` as the length of the string `"00:"`, but according to the [documentation of `g_strlcat`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strlcat) it should be the length of the destination buffer, here `str`, i.e. here `128`.

This leads to truncation in the string representation of the duration when the number of seconds result in the number of minutes being 0 (i.e. the else branch is taken), e.g. 3634 seconds -> "1:034" instead of "1:00:34"

The documentation of the return value of `g_strlcat` hints to this truncation:
> Returns
size of attempted result, which is MIN (dest_size, strlen (original dest)) + strlen (src), so if retval >= dest_size, truncation occurred.

I used the value `24` instead of `128` for consistency with the existing code, which uses `24` as the third argument in the rest of `convert_length_str`. I don't understand why `tmp` has that size, while `str` is of length `128`, considering that even if the third argument to `g_strlcat` was actually the size of the `src` instead of `dest`, the final string cannot be longer than 92 characters (4 `tmp` buffers, removing the nulls) excluding the terminating null = 93 in total.

I did not want to change all the `24`s to `128`s without further analysis or your opinion on the matter.